### PR TITLE
creates complex.lib and and testing to accompany it.

### PR DESCRIPTION
**Pull request by markcmoore**

### CODE REVIEW

File: address/.azure/azureup.yaml [hunk 1]
```diff
@@ -8,14 +8,15 @@ stages:
     jobs:
       - job:
         displayName: 'build::aspnet'
-        pool: Default
```
Comment (gpyofavv) [inline address/.azure/azureup.yaml]: You should not change this file from the base file in master

File: complex/.docker/dockerfile [hunk 2]
```diff
@@ -6,6 +6,7 @@ ARG DOTNET_VERSION=3.0
 ## stage - restore
 FROM mcr.microsoft.com/dotnet/core/sdk:${DOTNET_VERSION} as restore
 WORKDIR /src
+#COPY src/Revature.Complex.Api/*.csproj Revature.Complex.Api/
```
Comment (gpyofavv) [inline complex/.docker/dockerfile]: Remove this unnecessary comment

File: complex/src/Revature.Complex.Api/Revature.Complex.Api.csproj [hunk 3]
```diff
@@ -2,12 +2,27 @@
...
+    <UserSecretsId>7a5ab392-9d61-4399-9b92-ede92d5605f4</UserSecretsId>
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="3.0.0" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.Design" Version="3.0.0">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+    </PackageReference>
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="3.0.0" />
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Api/Revature.Complex.Api.csproj]: You don't need SqlServer!!! Or Tools.

File: complex/src/Revature.Complex.Lib/Interface/IRepository.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Collections.Generic;
+using Logic = Revature.Complex.Lib.Models;
+using System.Threading.Tasks;
+
+namespace Revature.Complex.Lib.Interface
+{
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Lib/Interface/IRepository.cs]: Interfaces must have documentation - if someone wanted to implement it, how would they know what it was expected to do?

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Revature.Complex.Lib.Models
+{
+    public class Amenity
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Lib/Models/Amenity.cs]: For Models (Logic, Entity, and Rest) you should have at least a good class description. Not necessarily a comment for every field/property.)

Comment (gpyofavv) [reply complex/src/Revature.Complex.Lib/Models/Amenity.cs]: This goes the same for all other models

File: complex/src/Revature.Complex.Lib/obj/Debug/netstandard2.1/Revature.Complex.Lib.AssemblyInfo.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+//------------------------------------------------------------------------------
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Lib/obj/Debug/netstandard2.1/Revature.Complex.Lib.AssemblyInfo.cs]: obj files should be gitignored!!!!!

Comment (gpyofavv) [reply complex/src/Revature.Complex.Lib/obj/Debug/netstandard2.1/Revature.Complex.Lib.AssemblyInfo.cs]: Same for all obj files.

File: housing/.azure/azureup.yaml [hunk 7]
```diff
@@ -8,7 +8,8 @@ stages:
     jobs:
```
Comment (gpyofavv) [inline housing/.azure/azureup.yaml]: Should be identical to master

File: address/.azure/azureup.yaml (hunk 1 above)
Comment (markcmoore) [reply address/.azure/azureup.yaml]: Seems all the green IS the version from the master. Which makes sense bc I copied it from master. I'm asking my team why they've changed them in their versions.

File: complex/src/Revature.Complex.Lib/Interface/IRepository.cs (hunk 4 above)
Comment (markcmoore) [reply complex/src/Revature.Complex.Lib/Interface/IRepository.cs]: I instructed Sam to put more commentary on these tomorrow. They will be fixed in the next merge.

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs (hunk 5 above)
Comment (markcmoore) [reply complex/src/Revature.Complex.Lib/Models/Amenity.cs]: I instructed Sam to put more commentary on these tomorrow. They will be fixed in the next merge.

File: address/.azure/azureup.yaml (hunk 1 above)
Comment (markcmoore) [reply address/.azure/azureup.yaml]: they didn't change them. They are correct in out copy. I don't know what this is from.

**Comment by gpyofavv** [comment]
Good progress on fixing, your azureup.yaml is still not the same as master (otherwise the pipeline would have succeeded). Get the rest of the stuff fixed and I'll recheck it in the morning.

**Comment by markcmoore** [comment]
Found a duplicate itemgroup somehow in the API. Once removed, all checks passed,

File: address/.azure/azureup.yaml [hunk 8]
```diff
@@ -223,7 +222,6 @@ stages:
               ARM_TENANT_ID: $(arm.tenant.id)
               CLOUDFLARE_EMAIL: $(cloudflare.email)
               CLOUDFLARE_TOKEN: $(cloudflare.token)
-
```
Comment (escalonn) [inline address/.azure/azureup.yaml]: no need for any diff to these pipeline files, restore the master branch's versions.

File: complex/src/Revature.Complex.Api/appsettings.json [hunk 9]
```diff
@@ -1,3 +1,4 @@
 {
-  "AllowedHosts": "*"
+  "ComplexDb": "Host=data;Database=room;Username=postgres;Password=Pass@word",
+  "StartDbCommand": "docker run -it --rm -p 11433:1433 -e SA_PASSWORD=Pass@word -e ACCEPT_EULA=Yes mcr.microsoft.com/mssql/server:2017-latest"
```
Comment (escalonn) [inline complex/src/Revature.Complex.Api/appsettings.json]: revert these changes. if you want to have these settings, put them in the gitignored appsettings.Development.json

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs (hunk 5 above)
Comment (escalonn) [inline complex/src/Revature.Complex.Lib/Models/Amenity.cs]: there's still no documentation on these domain models

File: address/.azure/azureup.yaml (hunk 8 above)
Comment (markcmoore) [reply address/.azure/azureup.yaml]: I C/P'd the masters versions in. I did this last night also. Lets see if it sticks this time.

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs (hunk 5 above)
Comment (markcmoore) [reply complex/src/Revature.Complex.Lib/Models/Amenity.cs]: documentation added.

File: complex/src/Revature.Complex.Api/appsettings.json (hunk 9 above)
Comment (markcmoore) [reply complex/src/Revature.Complex.Api/appsettings.json]: I commented them out. It still builds but I'm not sure it the service will be ok without them.

File: complex/src/Revature.Complex.Api/appsettings.json [hunk 10]
```diff
@@ -1,4 +1,5 @@
 {
-  "ComplexDb": "Host=data;Database=room;Username=postgres;Password=Pass@word",
-  "StartDbCommand": "docker run -it --rm -p 11433:1433 -e SA_PASSWORD=Pass@word -e ACCEPT_EULA=Yes mcr.microsoft.com/mssql/server:2017-latest"
+  "AllowedHosts": "*"
+//  "ComplexDb": "Host=data;Database=room;Username=postgres;Password=Pass@word",
+  //"StartDbCommand": "docker run -it --rm -p 11433:1433 -e SA_PASSWORD=Pass@word -e ACCEPT_EULA=Yes mcr.microsoft.com/mssql/server:2017-latest"
```
Comment (escalonn) [inline complex/src/Revature.Complex.Api/appsettings.json]: json does not technically allow comments, although asp.net will probably accept it.
like i said, this file should simply be restored to exactly what is already in master branch.

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs [hunk 11]
```diff
@@ -4,6 +4,9 @@
 
 namespace Revature.Complex.Lib.Models
 {
+  /// <summary>
+  /// This object defines an amenity. It have the amenity name and GUID (id).
```
Comment (escalonn) [inline complex/src/Revature.Complex.Lib/Models/Amenity.cs]: "have"?
we can see that it has those properties. useful documentation here would clarify with some examples, and indicate that some amenities are scoped to a particular room, while others apply to a whole complex at once.

File: complex/src/Revature.Complex.Lib/Models/AmenityComplex.cs [hunk 12]
```diff
@@ -4,6 +4,9 @@
 
 namespace Revature.Complex.Lib.Models
 {
+  /// <summary>
+  /// HTis object model defines a complex amenity. It has the amenity name and GUID (id)
```
Comment (escalonn) [inline complex/src/Revature.Complex.Lib/Models/AmenityComplex.cs]: HTis?

but there should not be these join-table entities in the domain layer at all. that is a data-access consideration.

Comment (markcmoore) [reply complex/src/Revature.Complex.Lib/Models/AmenityComplex.cs]: I clarified with Sam. This is a data model. The entity model is in the next PR.

File: address/.azure/azureup.yaml [hunk 13]
```diff
@@ -247,4 +247,3 @@ variables:
   - group: azure.vars
   - group: cloudflare.vars
   - group: sonarcloud.vars
-
```
Comment (gpyofavv) [inline address/.azure/azureup.yaml]: MARK WHY

File: complex/src/Revature.Complex.Api/Properties/launchSettings.json [hunk 14]
```diff
@@ -1,10 +1,8 @@
-{
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Api/Properties/launchSettings.json]: MARK YOU ARE KILLING ME

File: complex/src/Revature.Complex.Api/Startup.cs [hunk 15]
```diff
@@ -58,7 +58,7 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
       app.UseSwagger();
       app.UseSwaggerUI(c =>
       {
-        c.SwaggerEndpoint("/swagger/v1/swagger.json", "Revature Complex V1");
+          c.SwaggerEndpoint("/swagger/v1/swagger.json", "Revature Complex V1");
```
Comment (gpyofavv) [inline complex/src/Revature.Complex.Api/Startup.cs]: Indentation

File: complex/src/Revature.Complex.Lib/Models/Amenity.cs [hunk 16]
```diff
@@ -5,9 +5,10 @@
 namespace Revature.Complex.Lib.Models
 {
   /// <summary>
-  /// This object defines an amenity. It have the amenity name and GUID (id).
+  /// This object defines an amenity. It has the amenity name and GUID (id).
+  /// Some amenities are scoped to a particular room, while others apply to a whole complex.
   /// </summary>
-    public class Amenity
+  public class Amenity
     {
```
Comment (escalonn) [inline complex/src/Revature.Complex.Lib/Models/Amenity.cs]: indent

File: room/.docker/dockerfile [hunk 17]
```diff
@@ -6,7 +6,7 @@ ARG DOTNET_VERSION=3.0
 ## stage - restore
 FROM mcr.microsoft.com/dotnet/core/sdk:${DOTNET_VERSION} as restore
 WORKDIR /src
-COPY src/Revature.Room.Api/*.csproj Revature.Room.Api/
+COPY src/Revature.Room.Api.csproj Revature.Room.Api/
```
Comment (escalonn) [inline room/.docker/dockerfile]: the fateful tony diff that mark, greg, and i all let into master :elmo:
